MCR-2055 fix incorrect stream handling and file reading

rochpaul opened the pull request:
[Link to jira](https://mycore.atlassian.net/projects/MCR/issues/MCR-2055).

rochpaul committed:
MCR-2055 fix close stream on Files.list

rochpaul committed:
MCR-2055 use try-catch with resources for Files.list stream

rochpaul committed:
MCR-2055 use try catch resources to close stream correctly

rochpaul committed:
Merge branch 'master' into issues/MCR-2055-remove_commons_vfs2_windows_problems

rochpaul committed:
Merge branch 'master' into issues/MCR-2055-remove_commons_vfs2_windows_problems

rochpaul committed:
MCR-2055 getNumChildren() in MCRNode doesn't close Stream on Files.list

This causes Exceptions on file removement from derivate

rochpaul committed:
MCR-2055 findMaxID in MCRStore does not close stream correctly

Open stream on Files.list causes 'too many open files'

rochpaul committed:
MCR-2055 Reformat comitted Files with mycore-javastyle.xml

rochpaul committed:
MCR-2055 PathContent.asByteArray() causes lock files in windows

MyCoRe-Travis commented:
Something gone wrong on b84b83b1d5ce8f50ff33c3f7adc7f05afc6b5d9b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4575-pull_request-807

rochpaul committed:
MCR-2055 Rewrite PathContent.asByteArray()because of file locks

Have a look at http://www.mapdb.org/blog/mmap_files_alloc_and_jvm_crash/
Do not use memory memory mapped files in windows

rochpaul committed:
MCR-2055 revert MCRXMLMetadataManager

MyCoRe-Travis commented:
Something gone wrong on 8d82e64a9c5829de797fefc363fc248f2514c9d8! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4577-pull_request-807

rochpaul committed:
MCR-2055 Fix all not closed Files.list Streams in code
